packages/firestore/src/core/view.ts

```diff
@@ -403,9 +412,11 @@ export class View {
     }
   }
 
-  private updateLimboDocuments(): LimboDocumentChange[] {
+  private updateLimboDocuments(
+    waitForRequeryResult: boolean
+  ): LimboDocumentChange[] {
     // We can only determine limbo documents when we're in-sync with the server.
-    if (!this.current) {
+    if (waitForRequeryResult || !this.current) {
```

Comment:
Optional: Consider moving this `if` statement to the caller. IMO, it's more readable to just _not_ call a method named "updateLimboDocuments" if you don't want to "update limbo documents" (rather than calling the method with a boolean that will cause it to return immediately). It also avoids duplicating the default value of `[]` both here and at the singular call site.